index.js

```diff
@@ -1,20 +1,606 @@
-const readline = require('readline');
-const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+//Boilerplate code set up correctly - used to accept input from user
+const { ENETRESET } = require("constants");
+const { read } = require("fs");
+const { exit } = require("process");
+const readline = require("readline");
+const readlineInterface = readline.createInterface(
+  process.stdin,
+  process.stdout
+);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     readlineInterface.question(questionText, resolve);
   });
 }
 
-start();
+//------------------------------Player------------------------------//
+//defines an object to hold information about the player
+let player = {
+  name: "Player1",
+  inventory: [],
+  currentRoom: "",
+};
+
+//------------------------------Items------------------------------//
+//defines a class of items - constructors include
+// - name (full name of the item)
+// - descriptionTxt (text about the object when the player interacts with it)
+// - readTxt (text about the object when the player reads it)
+// - greetTxt (text about the object when the player greets it)
+// - takeable (boolean indicating if the object is takable or not)
+class checkInventory {
+  constructor(name, descriptionTxt, readTxt, greetTxt, takeable) {
+    this.name = name;
+    this.descriptionTxt = descriptionTxt;
+    this.readTxt = readTxt;
+    this.greetTxt = greetTxt;
+    this.takeable = takeable;
+  }
+  examine() {
+    return this.descriptionTxt;
+  }
+  read() {
+    return this.readTxt;
+  }
+
+  greet() {
+    return this.greetTxt;
+  }
+
+  take() {
+    if (this.takeable) {
+      //adds item to players inventory array
+      player.inventory.push(this.name);
+      return (
+        "You have picked up the " +
+        this.name +
+        ". You currently have: " +
+        player.inventory.join(", ")
+      );
+    } else {
+      return "You can't take the " + this.name + ".";
+    }
+  }
+
+  drop() {
+    //removes item from player inventory
+    player.inventory.splice(player.inventory.indexOf(this.name), 1); // index,how many item to be removed
```

Comment:
You also might want to add the item to the rooms inventory at this point as well so you can come back and pick it up later